packages/core/variables/_font-size.scss

```diff
@@ -26,142 +26,87 @@ $line-height-6: rem(48px);
 $line-height-7: rem(56px);
 $line-height-8: rem(72px);
 
-@mixin h1-heading {
-    @include h1-heading--desktop;
-    @include small-device {
-        @include h1-heading--mobile;
+$text-styles: (
+    "desktop/title-large": (
+        font-size: $font-size-12,
+        line-height: $line-height-8,
+        font-weight: $normal-weight,
+    ),
+    "compact/title-large": (
+        font-size: $font-size-10,
+        line-height: $line-height-6,
+        font-weight: $normal-weight,
+    ),
+    "desktop/title-small": (
+        font-size: $font-size-11,
+        line-height: $line-height-7,
+        font-weight: $normal-weight,
+    ),
+    "compact/title-small": (
+        font-size: $font-size-9,
+        line-height: $line-height-5,
+        font-weight: $normal-weight,
+    ),
+    "desktop/heading-large": (
+        font-size: $font-size-9,
+        line-height: $line-height-5,
+    ),
+    "compact/heading-large": (
+        font-size: $font-size-7,
+        line-height: $line-height-4,
+    ),
+    "desktop/heading-medium": (
+        font-size: $font-size-7,
+        line-height: $line-height-4,
+    ),
+    "compact/heading-medium": (
+        font-size: $font-size-5,
+        line-height: $line-height-3,
+    ),
+    "desktop/heading-small": (
+        font-size: $font-size-5,
+        line-height: $line-height-3,
+    ),
+    "compact/heading-small": (
+        font-size: $font-size-3,
+        line-height: $line-height-2,
+    ),
+    "desktop/lead": (
+        font-size: $font-size-8,
+        line-height: $line-height-5,
+    ),
+    "compact/lead": (
+        font-size: $font-size-6,
+        line-height: $line-height-4,
+    ),
+    "desktop/body": (
+        font-size: $font-size-4,
+        line-height: $line-height-4,
+    ),
+    "compact/body": (
+        font-size: $font-size-3,
+        line-height: $line-height-2,
+    ),
+    "desktop/small": (
+        font-size: $font-size-2,
+        line-height: $line-height-2,
+    ),
+    "compact/small": (
+        font-size: $font-size-2,
+        line-height: $line-height-2,
+    ),
+    "desktop/micro": (
+        font-size: $font-size-1,
+        line-height: $line-height-1,
+    ),
+    "compact/micro": (
+        font-size: $font-size-1,
+        line-height: $line-height-1,
+    ),
+);
+
+@mixin jkl-text-style($text-style) {
```

Comment:
Noen spørsmål om APIet her,

1. Er eneste grunnen til "desktop/small" syntaxen navngivningen i figma?
2. Stor sett overalt hvor det er satt opp responsiv typografi så er det samme element, i ulik utgave, feks "dekstop/lead" og "compact/lead", kunne vi ikke supplied ferdige mixins for det?
3. Burde ha feilhåntering i mixinen, veldig fort fort gjort med en liten typo her

Kunne det vært fornuftig med `@include jkl-text-size('desktop', 'lead')` som byggestein, og `@include jkl-text-size-responsive('lead')` feks ?

Reply:
Merket at det jeg egentlig vil ha er [JSSS](https://en.wikipedia.org/wiki/JavaScript_Style_Sheets)

Reply:
1. Jeg vurderte løsningen du foreslår med `jkl-text-style('desktop', 'body')`, men valgte å gjøre det likt som det står i Figma. Tanken var å få konseptet likt i koden og i Figma. Hvis designerne tenker på stilene som `desktop -> body` er nok forslaget ditt bedre. 
2. Problemet med en sånn mixin (som vi hadde tidligere) er at det ikke er åpenbart hva den gjør og hvordan den gjør det. Den introduserte bugs i flere av komponentene våre ved endringer pga. hvordan media queries rendres ut av Sass. Forslaget ditt med å kalle den `jkl-text-style--responsive` gjør det i hvert fall mer eksplisitt, så jeg er mer positiv til den 😊
3. Enig! Blir vel neppe fuzzy search, men kan i hvert fall legge inn en enkel håndtering som gir oversikt over gyldige verdier.